src/neo/SmartContract/InteropService.Storage.cs

```diff
@@ -165,6 +210,7 @@ private static bool Storage_Delete(ApplicationEngine engine)
                     };
                     if (engine.Snapshot.Storages.TryGet(key)?.IsConstant == true) return false;
                     engine.Snapshot.Storages.Delete(key);
+                    engine.TryAddUpdatedKey(key);
```

Comment:
If we don't consider reusing, is it possible to refund funds here. If so, which will greatly simplify. But I'm not sure.

Reply:
This is to prevent using refund feature to frequently add and delete the same key when one transaction is executing, and I am not quite sure whether such behavior should be discouraged. Maybe we should listen to neo-core's opinions later.

Reply:
We have to make sure here that the key was deleted, otherwise we can call the non-existing keys.

Reply:
Please don't mark as resolved, otherwise we can't track the reviews

Reply:
Sorry, maybe I have a misunderstanding. Do you mean  `engine.TryAddUpdatedKey(key);` may call a deleted key? It doesn't matter because it is only a cache for saving deleted or updated key.

Reply:
In order to prevent the occurrence of recycleReward for keys that have been deleted or do not exist, there is already a relevant check in `GetDeletedPrice`.
```C#
       var skeyValue = engine.Snapshot.Storages.TryGet(skey);
       if (skeyValue == null || skeyValue.Value == null || skeyValue.Value.Length == 0 || engine.HasUpdatedKey(skey))
           return 0_01000000;
```